src/api/me.py

```diff
@@ -76,3 +85,202 @@ async def delete_self(
         db: Database = Depends(get_db)
 ) -> None:
     await DatabaseHandler.delete_user(db.users, current_user['_id'])
+
+
+@router.get(
+    path='/wishlist',
+    response_model=PaginatedAlcohol,
+    status_code=status.HTTP_200_OK,
+    summary='Read user wishlist with pagination',
+    response_model_by_alias=False
+)
+async def get_wishlist(
+        limit: int = 10,
+        offset: int = 0,
+        db: Database = Depends(get_db),
+        current_user: UserDb = Depends(get_valid_user)
+) -> PaginatedAlcohol:
+    """
+    Show user wishlist with pagination
+    """
+    user_id = str(current_user['_id'])
+    alcohols = await UserWishlistHandler.get_user_wishlist_by_user_id(
+        limit, offset, db.user_wishlist, db.alcohols, user_id
+    )
+    return PaginatedAlcohol(
+        alcohols=alcohols,
+        page_info=PageInfo(
+            limit=limit,
+            offset=offset,
+            total=len(alcohols)
+        )
+    )
+
+
+@router.get(
+    path='/favourites',
+    response_model=PaginatedAlcohol,
+    status_code=status.HTTP_200_OK,
+    summary='Read user favourite alcohol list with pagination',
+    response_model_by_alias=False
+)
+async def get_favourites(
+        limit: int = 10,
+        offset: int = 0,
+        db: Database = Depends(get_db),
+        current_user: UserDb = Depends(get_valid_user)
+) -> PaginatedAlcohol:
+    """
+    Show user favourite alcohol list with pagination
+    """
+    user_id = str(current_user['_id'])
+    alcohols = await UserFavouritesHandler.get_user_favourites_by_user_id(
+        limit, offset, db.user_favourites, db.alcohols, user_id
+    )
+    return PaginatedAlcohol(
+        alcohols=alcohols,
+        page_info=PageInfo(
+            limit=limit,
+            offset=offset,
+            total=len(alcohols)
+        )
+    )
+
+
+@router.get(
+    path='/search_history',
+    response_model=PaginatedSearchHistory,
+    status_code=status.HTTP_200_OK,
+    summary='Read user search history list with pagination',
+    response_model_by_alias=False
+)
+async def get_search_history(
+        limit: int = 10,
+        offset: int = 0,
+        db: Database = Depends(get_db),
+        current_user: UserDb = Depends(get_valid_user)
+) -> PaginatedSearchHistory:
+    """
+    Show user search history alcohol list with pagination
+    """
+    user_id = str(current_user['_id'])
+    alcohols = await SearchHistoryHandler.get_user_search_history_user_id(
+        limit, offset, db.user_search_history, db.alcohols, user_id
+    )
+    return PaginatedSearchHistory(
+        alcohols=alcohols,
+        page_info=PageInfo(
+            limit=limit,
+            offset=offset,
+            total=len(alcohols)
+        )
+    )
+
+
+@router.delete(
+    path='/wishlist/{alcohol_id}',
+    status_code=status.HTTP_204_NO_CONTENT,
+    summary='Delete alcohol from wishlist'
+)
+async def delete_alcohol_form_wishlist(
```

Comment:
from* w innych miejscach to samo.
Zastanawiam się, czy nie lepiej byłoby zwracać zaktualizowane listy po operacjach na nich... 🤔 Pewnie pytanie do frontu co im bardziej pasuje